custom_components/sems/sensor.py

```diff
@@ -5,38 +5,567 @@
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-ibattery",
+                        f"Inverter {inverter['name']} Battery {idx} Current",
+                        path_to_battery + ["ibattery"],
+                        SensorDeviceClass.CURRENT,
+                        ELECTRIC_CURRENT_AMPERE,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-soc",
+                        f"Inverter {inverter['name']} Battery {idx} State of Charge",
+                        path_to_battery + ["soc"],
+                        SensorDeviceClass.BATTERY,
+                        PERCENTAGE,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-soh",
+                        f"Inverter {inverter['name']} Battery {idx} State of Health",
+                        path_to_battery + ["soh"],
+                        SensorDeviceClass.BATTERY,
+                        PERCENTAGE,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-bms_temperature",
+                        f"Inverter {inverter['name']} Battery {idx} BMS Temperature",
+                        path_to_battery + ["bms_temperature"],
+                        SensorDeviceClass.TEMPERATURE,
+                        TEMP_CELSIUS,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-bms_discharge_i_max",
+                        f"Inverter {inverter['name']} Battery {idx} BMS Discharge Max Current",
+                        path_to_battery + ["bms_discharge_i_max"],
+                        SensorDeviceClass.CURRENT,
+                        ELECTRIC_CURRENT_AMPERE,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-bms_charge_i_max",
+                        f"Inverter {inverter['name']} Battery {idx} BMS Charge Max Current",
+                        path_to_battery + ["bms_charge_i_max"],
+                        SensorDeviceClass.CURRENT,
+                        ELECTRIC_CURRENT_AMPERE,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                ]
+
+    if "hasPowerflow" in data and data["hasPowerflow"] and "powerflow" in data:
+        inverter_serial_number = get_home_kit_sn(data)
```

Comment:
Here you should add check for data["powerflow"]["isHomKit"]. If it is false there is no HomeKit sn and name.

Reply:
This falls through to the next statement if you don't have a homekit and we just call it powerflow

```python
if not has_existing_homekit_entity or inverter_serial_number is None:
            inverter_serial_number = "powerflow"
```

Reply:
But that will still create the homekit device right? It is confusing.